app/lib/actions/katello/repository/create_root.rb

```diff
@@ -3,21 +3,29 @@ module Katello
     module Repository
       class CreateRoot < Actions::EntryAction
         def plan(root, relative_path = nil)
-          begin
-            root.save!
-          rescue ActiveRecord::RecordInvalid => e
-            if root.is_container_push && e.message.include?("Container Repository Name") && e.message.include?("conflicts with an existing repository")
-              logger.warn("Skipping repository creation as container push repository already exists: #{root.container_push_name}")
-              return
+          repository = ::Katello::Repository.new(:environment => root.organization.library, :content_view_version => root.organization.library.default_content_view_version, :root => root)
```

Comment:
What do you think about splitting these changes out into `::Actions::Katello::Repository::CreateContainerPushRootRepository`?

The majority of the action now has code relating only to container push.

Reply:
That makes sense..Refactored the container push root creation into it's own action and reverted the CreateRoot to pre-container push state.